quelpa.el

```diff
@@ -1854,14 +1869,36 @@ nil."
   (when (quelpa-setup-p) ;if init fails we do nothing
     (let* ((quelpa-upgrade-p (if current-prefix-arg t quelpa-upgrade-p)) ;shadow `quelpa-upgrade-p'
            (quelpa-stable-p quelpa-stable-p) ;shadow `quelpa-stable-p'
-           (cache-item (if (symbolp arg) (list arg) arg)))
+           (cache-item (if (symbolp arg) (list arg) arg))
+           version)
       (quelpa-parse-plist plist)
       (quelpa-parse-stable cache-item)
-      (quelpa-package-install arg)
+      (when (setq version (quelpa-package-install arg))
+        (setq cache-item `(,(car cache-item)
+                           ,@(plist-put (cdr cache-item) :version version))))
       (quelpa-update-cache cache-item)))
   (quelpa-shutdown)
   (run-hooks 'quelpa-after-hook))
 
+;;;###autoload
+(define-minor-mode quelpa-mode
+  "Global minor mode for Quelpa."
+  :init-value t
+  :group 'quelpa
+  :global t
+  :lighter " Quelpa"
+  (if quelpa-mode
+      (progn
+        (advice-add #'package-delete :after
```

Comment:
I think we should either use a named function or use `define-advice`, because when users look at the advice list for `package-delete`, they won't know what the lambda does, especially when it's byte-compiled.  It's important that this advice be clearly described to users in case of problems.

Reply:
The `advice-add` has `name` property, so it will be shown to the user that function is being advice with `quelpa--package-delete`.

Reply:
`define-advice` seems to be more appropriate IMO as well